static/js/settings_bots.js

```diff
@@ -422,112 +421,121 @@ export function set_up() {
             value: user_id.toString(),
         }));
 
-        $("#edit_bot_modal").empty();
-        $("#edit_bot_modal").append(
-            render_edit_bot({
-                bot,
-                users_list,
-            }),
-        );
-        const avatar_widget = avatar.build_bot_edit_widget($("#settings_page"));
-        const form = $("#settings_page .edit_bot_form");
-        const image = li.find(".image");
-        const errors = form.find(".bot_edit_errors");
-
-        const opts = {
-            widget_name: "bot_owner",
-            data: users_list,
-            default_text: $t({defaultMessage: "No owner"}),
-            value: bot.owner_id,
-        };
-        const owner_widget = new DropdownListWidget(opts);
-
-        const service = bot_data.get_services(bot_id)[0];
-        if (bot.bot_type.toString() === OUTGOING_WEBHOOK_BOT_TYPE) {
-            $("#service_data").append(
-                render_settings_edit_outgoing_webhook_service({
-                    service,
-                }),
-            );
-            $("#edit_service_interface").val(service.interface);
-        }
-        if (bot.bot_type.toString() === EMBEDDED_BOT_TYPE) {
-            $("#service_data").append(
-                render_settings_edit_embedded_bot_service({
-                    service,
-                }),
-            );
-        }
+        const rendered_edit_bot = render_edit_bot({
+            bot,
+            users_list,
+        });
+        dialog_widget.launch({
+            html_heading: $t_html({defaultMessage: "Edit bot"}),
+            html_body: rendered_edit_bot,
+            id: "edit_bot_modal",
+            on_click: () => {
+                $(".edit_bot_form").trigger("submit");
+            },
+            post_render: edit_bot_post_render,
+            loading_spinner: true,
+        });
 
-        avatar_widget.clear();
+        function edit_bot_post_render() {
+            // This links the submit button to the edit bot form, so submitting this form
+            // by pressing Enter on an input element triggers a click on the submit button.
+            $("#edit_bot_modal .dialog_submit_button").attr("form", "edit_bot_form");
```

Comment:
Should this be a feature of `dialog_widget`, rather than a custom thing we do here?

Reply:
@timabbott  #20321